redis/Cargo.toml

```diff
@@ -65,11 +65,11 @@ tokio-native-tls = { version = "0.3", optional = true }
 async-native-tls = { version = "0.5", optional = true }
 
 # Only needed for rustls
-rustls = { version = "0.22", optional = true }
+rustls = { version = "0.23", optional = true, default-features = false, features = ["ring"] }
```

Comment:
I'd remove `features = ["ring"]` from here and instead add `"tokio-rustls/ring"` to the features enabled via `tls-rustls`.